### PR TITLE
python3.pkgs.w1thermsensor: init at 2.0.0

### PR DESCRIPTION
**Pull request by quentinmit**
###### Description of changes

This is a package for accessing 1-Wire temperature sensors.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
```
[root@bedroom-pi:~/nixpkgs]# w1thermsensor all
Got temperatures of 4 sensors:
  Sensor 1 (000002b1bcc5) measured temperature: 19.88 celsius
  Sensor 2 (0120541bbaa7) measured temperature: 9.5 celsius
  Sensor 3 (00000284c7a7) measured temperature: 20.81 celsius
  Sensor 4 (000002849de2) measured temperature: 27.19 celsius
```
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -12506,6 +12506,15 @@
...
   };
+  quentin = {
+    email = "quentin@mit.edu";
+    github = "quentinmit";
+    githubId = 115761;
+    name = "Quentin Smith";
+    keys = [{
+      fingerprint = "1C71 A066 5400 AACD 142E  B1A0 04EE 05A8 FCEF B697";
+    }];
+  };
```
Comment (mweinelt) [inline maintainers/maintainer-list.nix:12517]: Needs go do into its own commit. `maintainers: add quentin`.

File: pkgs/development/python-modules/w1thermsensor/default.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  postPatch = ''
+    sed -i 's/3\.5\.\*/3.5/' setup.py
+  '';
+
+  nativeBuildInputs = [
+    setuptools
+  ];
+
+  # Don't try to load the kernel module in tests.
+  W1THERMSENSOR_NO_KERNEL_MODULE = 1;
```
Comment (mweinelt) [inline pkgs/development/python-modules/w1thermsensor/default.nix:35]:
```suggestion
  env.W1THERMSENSOR_NO_KERNEL_MODULE = 1;
```

Explicitly make it an environment var like this.

File: pkgs/development/python-modules/w1thermsensor/default.nix [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  nativeCheckInputs = [
+    coverage
```
Comment (mweinelt) [inline pkgs/development/python-modules/w1thermsensor/default.nix:38]: We usually don't test coverage unless we really must. Can you remove this and patch out possible calls to `--cov*`.

File: pkgs/development/python-modules/w1thermsensor/default.nix [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    pytestCheckHook
+  ] ++ lib.optionals (pythonOlder "3.11") [
+    tomli
+  ];
+
+  pythonImportsCheck = [
+    "w1thermsensor"
+  ];
+
+  propagatedBuildInputs = [ aiofiles click ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/w1thermsensor/default.nix:50]: Please move this up below `nativeBuildInputs` and expand the list to one item per line,

File: pkgs/development/python-modules/w1thermsensor/default.nix [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  meta = with lib; {
+    description = "Python interface to 1-Wire temperature sensors";
+    longDescription = ''
+      A Python package and CLI tool to work with w1 temperature sensors like
+      DS1822, DS18S20 & DS18B20 on the Raspberry Pi, Beagle Bone and other
+      devices.
+    '';
+    homepage = "https://github.com/timofurrer/w1thermsensor";
+    license = licenses.mit;
+    maintainers = [ maintainers.quentin ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/w1thermsensor/default.nix:61]:
```suggestion
    maintainers = with maintainers; [ quentin ];
```

Easier to add/remove maintainers without repeating the qualifier.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (quentinmit) [reply maintainers/maintainer-list.nix:12517]: Does it also need a separate PR, or can I put two commits in this PR?

**Comment by mweinelt** [comment]
```
w1thermsensor> _________________________ test_repr_protocol[sensors0] _________________________
w1thermsensor> 
w1thermsensor> sensors = [{'id': '1', 'temperature': 20, 'type': <Sensor.DS18B20: 40>}]
w1thermsensor> 
w1thermsensor>     @pytest.mark.parametrize(
w1thermsensor>         "sensors",
w1thermsensor>         [
w1thermsensor>             # just a sensor
w1thermsensor>             ({"type": Sensor.DS18B20, "id": "1"},)
w1thermsensor>         ],
w1thermsensor>         indirect=["sensors"],
w1thermsensor>     )
w1thermsensor>     def test_repr_protocol(sensors):
w1thermsensor>         """Test the __repr__ protocol of a sensor instance"""
w1thermsensor>         # given
w1thermsensor>         sensor = W1ThermSensor()
w1thermsensor>         # when
w1thermsensor> >       sensor_copy = eval(repr(sensor))
w1thermsensor> 
w1thermsensor> tests/test_core.py:517: 
w1thermsensor> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
w1thermsensor> <string>:1: in <module>
w1thermsensor>     ???
w1thermsensor> /nix/store/44vp0f7j77ga3mkxwq3h3wvpd91qkn5i-python3.11-w1thermsensor-2.0.0/lib/python3.11/site-packages/w1thermsensor/core.py:150: in __init__
w1thermsensor>     self.BASE_DIRECTORY / (self.slave_prefix + self.id) / self.SLAVE_FILE
w1thermsensor> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
w1thermsensor> 
w1thermsensor> self = W1ThermSensor(sensor_type=40, sensor_id='1')
w1thermsensor> 
w1thermsensor>     @property
w1thermsensor>     def slave_prefix(self) -> str:
w1thermsensor>         """Returns the slave prefix for this temperature sensor"""
w1thermsensor> >       return "%s-" % hex(self.type.value)[2:]
w1thermsensor> E       AttributeError: 'int' object has no attribute 'value'
w1thermsensor> 
w1thermsensor> /nix/store/44vp0f7j77ga3mkxwq3h3wvpd91qkn5i-python3.11-w1thermsensor-2.0.0/lib/python3.11/site-packages/w1thermsensor/core.py:222: AttributeError
w1thermsensor> =========================== short test summary info ============================
w1thermsensor> FAILED tests/test_core.py::test_repr_protocol[sensors0] - AttributeError: 'int' object has no attribute 'value'
w1thermsensor> ======================== 1 failed, 144 passed in 1.05s =========================
```

Tests fail on python311. Best to report this upstream, if noone hasn't. We can disable the tests if it doesn't affect functionality, or disable the package on python311, if it does.

**Comment by quentinmit** [comment]
> Tests fail on python311. Best to report this upstream, if noone hasn't. We can disable the tests if it doesn't affect functionality, or disable the package on python311, if it does.

It looks like basic functionality works on python311:

```
[root@bedroom-pi:~/nixpkgs]# nix shell -L '.#python311.pkgs.w1thermsensor'
...
[root@bedroom-pi:~/nixpkgs]# w1thermsensor all
Got temperatures of 4 sensors:
  Sensor 1 (000002b1bcc5) measured temperature: 19.5 celsius
  Sensor 2 (0120541bbaa7) measured temperature: 5.81 celsius
  Sensor 3 (00000284c7a7) measured temperature: 20.69 celsius
  Sensor 4 (000002849de2) measured temperature: 29.0 celsius
```

So, I've updated the PR to disable the tests on 3.11 for now and filed https://github.com/timofurrer/w1thermsensor/issues/116 for the test failure.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (mweinelt) [reply maintainers/maintainer-list.nix:12517]: Fine like this.

File: pkgs/development/python-modules/w1thermsensor/default.nix [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  # Don't try to load the kernel module in tests.
+  env.W1THERMSENSOR_NO_KERNEL_MODULE = 1;
+
+  nativeCheckInputs = [
+    pytest-mock
+    pytest-asyncio
+  ] ++ lib.optionals (pythonOlder "3.11") [
+    tomli
+    # Tests for 2.0.0 currently fail on python3.11
+    # https://github.com/timofurrer/w1thermsensor/issues/116
```
Comment (mweinelt) [inline pkgs/development/python-modules/w1thermsensor/default.nix:48]: Would be better to set `doCheck = pythonOlder "3.11";` instead.

Comment (quentinmit) [reply pkgs/development/python-modules/w1thermsensor/default.nix:48]: If I do that, I would lose the import test on 3.11, no? It seems preferable to me to keep that, since it's still passing.

Comment (quentinmit) [reply pkgs/development/python-modules/w1thermsensor/default.nix]: Ah, I didn't realize that they were independent of `doCheck`. Thanks.  I've updated the PR to use `doCheck` as you suggested.
